api/src/routing.js

```diff
@@ -150,25 +152,58 @@ app.use(compression());
 
 // TODO: investigate blocking writes to _users from the outside. Reads maybe as well, though may be harder
 //       https://github.com/medic/medic/issues/4089
+const root = (req, res) => {
+  if (('deviceID' in req.query) || req.is('application/json')) {
+    // couchdb request - let it go
+    return proxy.web(req, res);
+  }
+  let redirectUrl = '/';
+  const startPosition = req.url.indexOf('#');
+  if (startPosition !== -1) {
+    redirectUrl = redirectUrl.concat(req.url.slice(startPosition));
+  }
+  res.redirect(redirectUrl);
+};
 
 app.get('/', function(req, res) {
   if (req.headers.accept === 'application/json') {
     // couchdb request - let it go
-    proxy.web(req, res);
+    return proxy.web(req, res);
   } else {
-    // redirect to the app path - redirect to _rewrite
-    res.redirect(appPrefix);
+    res.sendFile(path.join(STATIC_RESOURCE_DESTINATION, 'templates/inbox.html'));
   }
 });
+app.get(appPrefix, root);
 
-/*
-To facilitate service worker prefetch on Chrome <66, serve a version of the app which does not require authentication
-*/
-app.get(appPrefix, (req, res, next) => {
-  if ('_sw-precache' in req.query) {
-    return res.sendFile(path.join(__dirname, 'extracted-resources/templates/inbox.html'));
+app.all(/^\/medic$/, (req, res, next) => {
+  if (environment.db === 'medic') {
```

Comment:
This is almost always going to be true - API is always pointing to the medic database and it's almost always called "medic". I think the only time this isn't true is when running e2e tests, but it's really dangerous to have code that's only run for e2e tests because then you're not testing actual code!

What are you trying to do here?